java-checks/src/main/java/org/sonar/java/checks/tests/AbstractJUnit5NotCompliantModifierChecker.java

```diff
@@ -65,23 +71,43 @@ public void visitNode(Tree tree) {
     if (classTree.symbol().isAbstract()) {
       return;
     }
-    List<MethodTree> methods = classTree.members().stream()
+
+    List<MethodTree> junit5ClassMethods = new ArrayList<>();
+    List<MethodTree> junit5InstanceMethods = new ArrayList<>();
+    List<MethodTree> nonJunit5Methods = new ArrayList<>();
+    classTree.members().stream()
       .filter(member -> member.is(Tree.Kind.METHOD))
       .map(MethodTree.class::cast)
-      .collect(Collectors.toCollection(ArrayList::new));
-
-    List<MethodTree> testMethods = methods.stream()
-      .filter(UnitTestUtils::hasJUnit5TestAnnotation)
-      .filter(AbstractJUnit5NotCompliantModifierChecker::isNotOverriding)
-      .toList();
+      .forEach(method -> {
+        if (UnitTestUtils.hasJUnit5TestAnnotation(method) || UnitTestUtils.hasJUnit5InstanceLifecycleAnnotation(method)) {
+          if (isNotOverriding(method)) {
+            junit5InstanceMethods.add(method);
+          }
+        } else if (UnitTestUtils.hasJUnit5ClassLifecycleAnnotation(method)) {
+          if (isNotOverriding(method)) {
+            junit5ClassMethods.add(method);
+          }
+        } else {
+          nonJunit5Methods.add(method);
+        }
+      });
+
+    raiseIssueOnMethods(junit5ClassMethods, ModifierScope.CLASS_METHOD);
+    raiseIssueOnMethods(junit5InstanceMethods, ModifierScope.INSTANCE_METHOD);
+    if (!junit5InstanceMethods.isEmpty()) {
```

Comment:
Can we extract this for readability?
```suggestion
    boolean classHasJunit5InstanceMethods = !junit5InstanceMethods.isEmpty();
    if (classHasJunit5InstanceMethods) {
```